src/measure_theory/function/locally_integrable.lean

```diff
@@ -9,35 +9,124 @@ import measure_theory.integral.integrable_on
 # Locally integrable functions
 
 A function is called *locally integrable* (`measure_theory.locally_integrable`) if it is integrable
-on a neighborhood of every point.
+on a neighborhood of every point. More generally, it is *locally integrable on `s`* if it is
+locally integrable on a neighbourhood within `s` of any point of `s`.
 
-This file contains properties of locally integrable functions and integrability results
+This file contains properties of locally integrable functions, and integrability results
 on compact sets.
 
 ## Main statements
 
 * `continuous.locally_integrable`: A continuous function is locally integrable.
-
+* `continuous_on.locally_integrable_on`: A function which is continuous on `s` is locally
+  integrable on `s`.
 -/
 
 open measure_theory measure_theory.measure set function topological_space
 open_locale topology interval
 
 variables {X Y E R : Type*} [measurable_space X] [topological_space X]
 variables [measurable_space Y] [topological_space Y]
-variables [normed_add_comm_group E] {f : X → E} {μ : measure X}
+variables [normed_add_comm_group E] {f : X → E} {μ : measure X} {s : set X}
 
 namespace measure_theory
 
-/-- A function `f : X → E` is locally integrable if it is integrable on a neighborhood of every
+section locally_integrable_on
+
+/-- A function `f : X → E` is *locally integrable on s*, for `s ⊆ X`, if for every `x ∈ s` there is
+a neighbourhood of `x` within `s` on which `f` is integrable. (Note this is, in general, strictly
+weaker than local integrability with respect to `μ.restrict s`.) -/
```

Comment:
could you prove the lemma that local integrability with respect to `μ.restrict s` implies local integrability on `s`?

Reply:
Sure, no problem. The converse is also true if `s` is closed, do you think it'd be useful to have that lemma as well?

Reply:
Yes, that would be a nice lemma to have.